nucypher/blockchain/eth/registry.py

```diff
@@ -376,20 +379,26 @@ def search(self, beneficiary_address: str = None, contract_address: str = None):
         if beneficiary_address:
             try:
                 contract_data = allocation_data[beneficiary_address]
+                result = contract_data
             except KeyError:
                 raise self.UnknownBeneficiary
 
         elif contract_address:
             records = list()
             for beneficiary_address, contract_data in allocation_data.items():
-                contract_address, contract_abi = contract_data['address'], contract_data['abi']
-                records.append(dict(address=contract_address, abi=contract_abi))
+                if contract_address == contract_data[0]:
```

Comment:
It's a good idea, but since it should involve the same change to the contract registry, besides the allocation registry, I think it falls out of the scope of this PR.